source/backend/api/Services/DocumentQueueService.cs

```diff
@@ -243,12 +243,6 @@ public async Task<PimsDocumentQueue> Upload(PimsDocumentQueue documentQueue)
                 UpdateDocumentQueueStatus(databaseDocumentQueue, DocumentQueueStatusTypes.PIMS_ERROR);
                 return databaseDocumentQueue;
             }
-            else if (relatedDocument?.MayanId != null && relatedDocument?.MayanId > 0)
```

Comment:
This is incorrect, in addition to having a mayanId, we need to wait for the filedetail to be available.